packages/scripts/src/monitoring-dvm2.0/simulateVoting.ts

```diff
@@ -0,0 +1,663 @@
+// This script verifies that basic voting functionality works correctly in the VotingV2 contract.
+// It is adapted from original DVM2.0 upgrade tests in packages/scripts/src/upgrade-tests/voting2/3_SimulateVoting.ts
+// intended to be run as regular health checks on the VotingV2 contract.
+// This script can be run on a mainnet or goerli fork, with the following command and environment variables:
+// CUSTOM_NODE_URL=https://<goerli OR mainnet>.infura.io/v3/<YOUR-INFURA-KEY> \
+// yarn hardhat run ./src/monitoring-dvm2.0/simulateVoting.ts
+
+import hre from "hardhat";
+import { strict as assert } from "assert";
+
+const { formatBytes32String, formatEther, parseEther, parseUnits, toUtf8Bytes } = hre.ethers.utils;
+
+import { BigNumber, BigNumberish, BytesLike, Signer } from "ethers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import { OptimisticOracleRequestStatesEnum, PriceRequestStatusEnum } from "@uma/common";
+import { OptimisticOracleV2Ethers, StoreEthers } from "@uma/contracts-node";
+
+import { REQUIRED_SIGNER_ADDRESSES, SECONDS_PER_DAY } from "../utils/constants";
+import { getContractInstance } from "../utils/contracts";
+import { getForkChainId, increaseEvmTime } from "../utils/utils";
+import {
+  commitVote,
+  createCommittedVote,
+  getUniqueVoters,
+  getVotingContracts,
+  revealVote,
+  unstakeFromStakedAccount,
+  VotingPriceRequest,
+} from "../utils/votingv2-utils";
+
+interface PriceRequestData {
+  originalAncillaryData: BytesLike;
+  proposedPrice: BigNumberish;
+  priceRequest: VotingPriceRequest;
+}
+interface RewardTrackers {
+  lastUpdateTimestamp: BigNumber;
+  staked: [BigNumber, BigNumber, BigNumber];
+  unclaimedRewards: [BigNumber, BigNumber, BigNumber];
+}
+
+const foundationAddress = REQUIRED_SIGNER_ADDRESSES.foundation;
+const accountWithUma = REQUIRED_SIGNER_ADDRESSES.account_with_uma;
+
+const zeroBigNumber = hre.ethers.BigNumber.from(0);
+
+// Constants hardcoded in the SlashingLibrary, needs to be updated here upon change.
+const wrongVoteSlashPerToken = parseEther("0.001");
+const noVoteSlashPerToken = parseEther("0.001");
+
+// Initial voter balances relative to GAT.
+const voter1RelativeGatFunding = parseEther("1.1"); // Voter 1 reaches GAT and SPAT.
+const voter2RelativeGatFunding = parseEther("0.55");
+const voter3RelativeGatFunding = parseEther("0.5");
+
+// Tested price identifier should be whitelisted.
+const priceIdentifier = formatBytes32String("NUMERICAL");
+
+// Ensure VotingV2 contract has at least this much total stake in Wei so that rewardsPerTokenStored do not overflow.
+const minimumStakedAmount = hre.ethers.BigNumber.from("1000");
```

Comment:
Finally we opted to stake `0.000000000000001 UMA` instead of `1 UMA` right? Please can you explain this amount?

Reply:
Setting it to 1 UMA interfered with slash amount verification in Step 20. Thus, 1000 wei was experimentally found to be minimally sufficient for not triggering overflow while at the same time keeping slash amount rounding errors within 10 wei.
Pushed this comment in the latest commit.